src/libData/AccountData/Account.cpp

```diff
@@ -146,16 +232,65 @@ vector<string> Account::GetKeys()
     return ret;
 }
 
-vector<string> Account::GetStorage(string _k)
+vector<string> Account::GetStorage(string _k) const
 {
+    if (!isContract())
+        return {};
     dev::RLP rlp(m_storage[_k]);
+    // mutable, type, value
     return {rlp[0].toString(), rlp[1].toString(), rlp[2].toString()};
 }
 
-string Account::GetStorageValue(string _k) { return GetStorage(_k)[2]; }
+Json::Value Account::GetStorageJson() const
+{
+    Json::Value root;
+    vector<string> keys = GetKeys();
+    if (keys.empty())
+    {
+        root = Json::arrayValue;
+    }
+    for (auto k : keys)
+    {
+        vector<string> v = GetStorage(k);
+        if (v[0] == "False")
+        {
+            continue;
+        }
+        Json::Value item;
+        item["vname"] = k;
+        item["type"] = v[1];
+        if (v[1] == "Map")
```

Comment:
I think we may another special type ADT. We are trying to finalize that. Once it is done, we may need to update this part of the code.

Reply:
Okay